elenacaseyroby/YoutubeMusicLib
# pulls full album out and years out of titles

# What

pulls the following from last fm: genre (tags in last fm), start year, end year, city
# Why

so we have more metadata to play with
# Test
- update database
- listen to a few tracks on youtube (not full albums)
- check artists and vids_genres table to see if data was posted.
- merge!
# SQL

```
ALTER TABLE videos
ADD COLUMN track_num int(5) DEFAULT NULL;

ALTER TABLE albums
DROP COLUMN track_num;

ALTER TABLE cities
ADD COLUMN city_or_state varchar(100) DEFAULT NULL;

UPDATE cities
SET city_or_state = name;

ALTER TABLE cities
DROP COLUMN `name`;


```
# known bugs:
- lastfm.js not replacing all "," with spaces when it makes
  the artistandmatch string so sometimes match will be an artist name
  instead of an integer
- not sure why but sometimes youtube_id and genre don't get insterted into the vids_genres table correctly and since they are both foreign keys, it causes the insert to fail so the track's genres aren't stored.
